Sources/Controls/UIBarButtonItem+Combine.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  UIBarButtonItem+Combine.swift
+//  CombineCocoa
+//
+//  Created by Shai Mishali on 12/08/2019.
+//  Copyright © 2019 Shai Mishali. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+public extension UIBarButtonItem {
+  /// A publisher which emits whenever this UIBarButtonItem is tapped.
+  var tapPublisher: AnyPublisher<Void, Never> {
+    return TapPublisher(barButtonItem: self).eraseToAnyPublisher()
+  }
+}
+
+private extension UIBarButtonItem {
+  struct TapPublisher: Publisher {
+    public typealias Output = Void
+    public typealias Failure = Never
+
+    weak private var barButtonItem: UIBarButtonItem?
+
+    public init(barButtonItem: UIBarButtonItem) {
+      self.barButtonItem = barButtonItem
+    }
+
+    public func receive<S: Subscriber>(subscriber: S) where S.Failure == Failure, S.Input == Output {
+      guard let barButtonItem = barButtonItem else {
```

Comment:
> Why weak, var and optional? 🤔
> I think private let barButtonItem: UIBarButtonItem should be okay.

You wouldn't need to `guard` afterwards.

Reply:
Same as above